tests/integrationv2/test_sni_match.py

```diff
@@ -0,0 +1,76 @@
+import copy
+import pytest
+
+from configuration import available_ports, MULTI_CERT_TEST_CASES, PROVIDERS, PROTOCOLS
+from common import ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version
+
+
+def filter_cipher_list(*args, **kwargs):
+    """
+    The framework normally filters out ciphers that are not supported by the chosen
+    protocol. That doesn't happen in this test because of the unique way ciphers are
+    grouped for the multi certificate tests.
+
+    This function handles that unique grouping.
+    """
+    protocol = kwargs.get('protocol')
+    cert_test_case = kwargs.get('cert_test_case')
+
+    lowest_protocol_cipher = min(cert_test_case.client_ciphers, key=lambda x: x.min_version)
+    if protocol < lowest_protocol_cipher.min_version:
+        return True
+
+    return False
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.uncollect_if(func=filter_cipher_list)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("protocol", [Protocols.TLS13, Protocols.TLS12], ids=get_parameter_name)
+@pytest.mark.parametrize("cert_test_case", MULTI_CERT_TEST_CASES)
+def test_sni_match(managed_process, provider, protocol, cert_test_case):
+    host = "localhost"
+    port = next(available_ports)
+
+    client_options = ProviderOptions(
```

Comment:
Is it important to send some data after the handshake just to make sure that everything is working correctly?

Reply:
I didn't do that in this test because we are verifying data that is transferred entirely in the handshake. There happy path integration test verifies that data can be sent over the negotiated connection. #1844 describes another test that will send very large data chunks over a connection.